edittypes/tokenizer.py

```diff
@@ -0,0 +1,49 @@
+import re
+import string
+#Universal regex for punctuations
+
+class Tokenizer:
+
+    def __init__(self, english_unicode, non_english_unicode):
+        self.english_punc_regex = r'[{0}]'.format(re.escape(string.punctuation))
+        self.english_unicode = english_unicode
+        self.non_english_unicode =  non_english_unicode
+
+    def get_punctuations(self, text):
+        #get ellipsis
+        ellipsis = re.findall(r'\.{3,}',text)
+        text = re.sub(r'\.{3,}','',text)
+
+        #Get other punctuations
+        english_punc_regex = re.findall(self.english_punc_regex, text)
+
+        english_unicode_pattern = re.compile(self.english_unicode, re.UNICODE)
+        english_unicode_regex = re.findall(english_unicode_pattern, text)
+
+        non_english_unicode_pattern = re.compile(self.non_english_unicode, re.UNICODE)
+        non_english_punc_regex = re.findall(non_english_unicode_pattern, text)
+
+        return ellipsis + english_punc_regex +english_unicode_regex + non_english_punc_regex
+
+
+    def get_whitespace(self, text):
+        whitespace = re.findall(r'[\n\r\s]+',text)
+        return whitespace
+    
+
+    def get_words(self, text):
+        word_list = re.findall(r'\w+',text)
+        return word_list
+
+    def get_sentences(self, text):
+        sentences = re.split(r'[!?]+|(?<!\.)\.(?!(?<=\d.)\d)(?!\.)', text)
+        sentences = sentences[:-1]
+        return sentences
+
+    def tokenize_and_count(self, text):
```

Comment:
I think this is fine for now for testing but just FYI that when it comes to detecting changes in later PRs, we'll want to do more than count the length. For example, adding a space next to another space wouldn't change the length of the whitespace array, so probably wouldn't show up as a change. Instead, I imagine we'll want to convert these into dictionaries of counts -- e.g., `{' ': 6, '\n':1}` -- to compare.

Reply:
Would these apply to paragraphs and sentences as well?

Reply:
Hmm...good question. My initial thinking is that we build a counts dictionary for each of these categories (whitespace, words, sentences, paragraphs) and then take the difference between previous and current.

Something I'm struggling with is how to count for changes. For example, say the text starts with `Sentence with three spaces.\n\nNew paragraph and another five spaces.` and goes to `Sentence with three spaces. No new paragraph so six spaces.`
* The whitespace counts dictionaries would be `{' ':8, '\n\n':1}` and `{' ':9}`. Is this represented as `{'Whitespace': {'change':2}}` (difference gives `{' ':1, '\n\n':-1}` whose absolute values sum to 2)? If we tracked whitespace as closely as we do the other node though, this would be `{'Whitespace': {'change':1}}` because it was just one segment of whitespace (`\n\n`) becoming another segment (` `).
* The difference between the words would be `{'New':-1, 'and':-1, 'another':-1, 'five':-1, 'No':1, 'new':1, 'so':1, 'six':1}`. Is this a change of 8 (sum of absolute values) or a change of 4 (summing up all the deleted words (4) and all the added words (4) and taking the maximum of the two)?
* The difference between sentences would show one deleted and one added. Same as above -- is this one sentence changed or two?
* The difference in paragraphs would show three different paragraphs where two were deleted and one was added. Same as above -- is this three paragraphs changed or two?

Reply:
Thanks @geohci for the detailed explanation. I have more clarity now.